src/libAtomVM/term.c

```diff
@@ -680,6 +680,49 @@ TermCompareResult term_compare(term t, term other, TermCompareOpts opts, GlobalC
     return result;
 }
 
+void term_get_function_mfa(term fun, Context *ctx, term *m, term *f, term *a)
+{
+    TERM_DEBUG_ASSERT(term_is_fun(fun));
+
+    const term *boxed_value = term_to_const_term_ptr(fun);
+    if (term_is_external_fun(fun)) {
+        if (m != NULL) {
+            *m = boxed_value[1];
+        }
+        if (f != NULL) {
+            *f = boxed_value[2];
+        }
+        if (a != NULL) {
+            *a = boxed_value[3];
+        }
+        return;
+    }
+
+    Module *module = (Module *) boxed_value[1];
+    if (m != NULL) {
+        *m = module_get_name(module);
+    }
+    if (f != NULL) {
+        uint32_t fun_index = term_to_int32(boxed_value[2]);
+
+        uint32_t label, arity, n_freeze;
+        module_get_fun(module, fun_index, &label, &arity, &n_freeze);
+
+        AtomString fun_name = NULL;
+        bool has_local_name = module_get_function_from_label(module, label, &fun_name, (int *) &arity, ctx->global);
+
+        *f = has_local_name ? globalcontext_make_atom(ctx->global, fun_name) : term_nil();
+    }
+    if (a != NULL) {
+        uint32_t fun_index = term_to_int32(boxed_value[2]);
+
+        uint32_t label, arity_val, n_freeze;
+        module_get_fun(module, fun_index, &label, &arity_val, &n_freeze);
+
+        *a = term_make_maybe_boxed_int64(arity_val, &ctx->heap);
```

Comment:
For the sake of simplicity we might make the assumption that the fun takes less than 2^28 parameters, so we remove any boxed integer from here and we don't need any access to a Heap.

Reply:
Actually Erlang supports maximum of 255 parameters so we can fit it in `int11`. Good catch, I shouldn't heap-alloc it in the first place.